Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/resultavailable/SubmissionTestResultAvailableViewModel.kt

```diff
@@ -42,43 +42,41 @@ class SubmissionTestResultAvailableViewModel @AssistedInject constructor(
             override fun onTEKAvailable(teks: List<TemporaryExposureKey>) {
                 Timber.d("onTEKAvailable(teks.size=%d)", teks.size)
                 autoSubmission.updateMode(AutoSubmission.Mode.MONITOR)
-
+                showKeysRetrievalProgress.postValue(false)
                 routeToScreen.postValue(
                     SubmissionTestResultAvailableFragmentDirections
                         .actionSubmissionTestResultAvailableFragmentToSubmissionTestResultConsentGivenFragment()
                 )
-
-                showKeysRetrievalProgress.postValue(false)
             }
 
             override fun onTEKPermissionDeclined() {
                 Timber.d("onTEKPermissionDeclined")
+                showKeysRetrievalProgress.postValue(false)
```

Comment:
I'm wondering if it made sense to implement some sort of time-out for the dialog (as there is a submission time-out as well) or make the progress dialog dismissable. If we did, the user will not get stuck in that scenario.

Reply:
For submission dialog I did not see any timeout.
There are many questions for introducing a timeout , we can discuss it and create a task for it ,but it is out of scope here. I don't like the whole thing ,it was not introduced because the tester is just using a tool that is not supposed to reflect what is happening on an end user device

Reply:
thx. I agree with it being another task